### PR TITLE
TSK-1984 Refactor TaskCommentAccTests to use Test-API

### PR DESCRIPTION
**Pull request by arolfes**
[Sonar](https://sonarcloud.io/summary/new_code?id=arolfes_taskana&branch=TSK-1984-refactor-TaskCommentAccTests)

---
Release Notes:

```

```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+  @WithAccessId(user = "admin")
+  @BeforeAll
+  void setup() throws Exception {
+    defaultClassification =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    defaultWorkbasket =
+        DefaultTestEntities.defaultTestWorkbasket().buildAndStore(workbasketService);
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasket.getId())
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:90]: Add new WorkbasketPermission.READTASKS permission here

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    taskService.deleteTaskComment(comment1.getId());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(1);
+  }
+
+  @WithAccessId(user = "user-1-2", groups = "user-1-1") // to read comments
+  @Test
+  void should_FailToDeleteTaskComment_When_UserHasNoAuthorization() throws Exception {
+    assertThatThrownBy(() -> taskService.deleteTaskComment(comment1.getId()))
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:111]: Thrown Exception should be NotAuthorizedOnTaskCommentException, and follow the testing guideline of Exception

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    taskService.deleteTaskComment(comment1.getId());
+
+    List<TaskComment> taskCommentsAfterDeletion = taskService.getTaskComments(task1.getId());
+    assertThat(taskCommentsAfterDeletion).hasSize(1);
+  }
+
+  @WithAccessId(user = "admin")
+  @WithAccessId(user = "taskadmin")
+  @TestTemplate
+  void should_DeleteTaskComment_When_NoExplicitPermissionsButUserIsInAdministrativeRole()
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:131]: This test seems similar to the should_DeleteTaskComment_When_UserIsInAdministrativeRole

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  @WithAccessId(user = ACCESS_USER_1_1_ID)
+  @Test
+  void should_FailToDeleteTaskComment_When_TaskCommentIdIsNull() throws Exception {
+    assertThatThrownBy(() -> taskService.deleteTaskComment(null))
+        .isInstanceOf(InvalidArgumentException.class);
+  }
+
+  @WithAccessId(user = ACCESS_USER_1_1_ID)
+  @Test
+  void should_FailToDeleteTaskComment_When_TaskCommentIsNotExisting() throws Exception {
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/delete/DeleteTaskCommentAccTest.java:157]: Follow the testing guideline

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java [hunk 5]
```diff
@@ -0,0 +1,460 @@
...
+    userWithName.setLastName("Mustermann");
+    userWithName.setFullName("Max Mustermann");
+    userService.createUser(userWithName);
+
+    defaultClassification =
+        DefaultTestEntities.defaultTestClassification().buildAndStore(classificationService);
+    defaultWorkbasket =
+        DefaultTestEntities.defaultTestWorkbasket().buildAndStore(workbasketService);
+    WorkbasketAccessItemBuilder.newWorkbasketAccessItem()
+        .workbasketId(defaultWorkbasket.getId())
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java:117]: Add the new WorkbasketPermission.READTASKS permission

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java [hunk 6]
```diff
@@ -0,0 +1,460 @@
...
+          taskService.createTaskCommentQuery().taskIdIn(task1.getId()).list();
+      assertThat(comments).hasSize(3);
+    }
+
+    @WithAccessId(user = "user-1-2")
+    @Test
+    void should_ThrowException_When_NotAuthorizedToReadTaskComments() {
+
+      ThrowingCallable call =
+          () -> taskService.createTaskCommentQuery().taskIdIn(task1.getId()).list();
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java:150]: Follow testing guideline, check also if fields are set correctly

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java [hunk 7]
```diff
@@ -0,0 +1,460 @@
...
+
+      assertThat(taskComments).hasSize(1);
+      assertThat(taskComments.get(0)).extracting(TaskComment::getCreatorFullName).isNull();
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class TaskCreatorFullNameTaskCommentsWithoutAdditionalUserInfo
+      implements TaskanaEngineConfigurationModifier {
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java:442]: TaskanaEngineConfigurationModifier has been deprecated and changed to TaskanaConfigurationModifier

File: lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java [hunk 8]
```diff
@@ -0,0 +1,460 @@
...
+
+    @TaskanaInject TaskService taskService;
+
+    @Override
+    public void modify(TaskanaEngineConfiguration taskanaEngineConfiguration) {
```
Comment (jamesrdi) [inline lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java:447]: TaskanaEngineConfiguration has been deprecated, use TaskanaConfiguration.Builder instead

**Review by jamesrdi — CHANGES_REQUESTED** [review]

**Review by jamesrdi — CHANGES_REQUESTED** [review]
Please rebase this PR

**Comment by arolfes** [comment]
Hello @jamesrdi ,
Thank you for reviewing this old PR.
Unfortunately, I don't have time at the moment to implement the changes you've rightly suggested.

If you have more time, please throw the PR away and start from scratch. Knowing my current workload, I won't be able to do it until at least Christmas.

Best Regards Alex

**Comment by jamesrdi** [comment]
Ok @arolfes I will take care of this PR :)

**Comment by arolfes** [comment]
> Ok @arolfes I will take care of this PR :)

Thank you, I close this PR now

**Pull request closed without merging**